BoxContentSDK/BoxContentSDK/Protocols/BOXContentCacheClientProtocol.h

```diff
@@ -0,0 +1,54 @@
+//
+//  BOXContentCacheClientProtocol.h
+//  BoxContentSDK
+//
+//  Created by Scott Liu on 10/13/15.
+//  Copyright © 2015 Box. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+@class BOXBookmarkCommentsRequest;
+@class BOXCommentRequest;
+@class BOXCommentAddRequest;
+@class BOXCommentDeleteRequest;
+@class BOXCommentUpdateRequest;
+@class BOXFileCommentsRequest;
+
+@protocol BOXContentCacheClientProtocol <NSObject>
+
+#pragma mark - Comments
+
+- (void)cacheBookmarkCommentsRequest:(BOXBookmarkCommentsRequest *)request
+                        withComments:(NSArray *)comments;
+
+- (void)retrieveCacheForBookmarkCommentsRequest:(BOXBookmarkCommentsRequest *)request
+                                     completion:(BOXObjectsArrayCompletionBlock)completionBlock;
+
+- (void)cacheCommentsRequest:(BOXCommentRequest *)request
+                 withComment:(BOXComment *)comment;
+
+- (void)retrieveCacheForCommentRequest:(BOXCommentRequest *)request
+                            completion:(BOXCommentBlock)completionBlock;
+
+- (void)cacheAddCommentRequest:(BOXCommentAddRequest *)request
+                   withComment:(BOXComment *)comment;
```

Comment:
we should have error as an input parameter as well for all the interface methods. Depending on the error, e.g. unauthorized, the cache client needs to delete the cache entries or not.